client/starwhale/core/dataset/tabular.py

```diff
@@ -213,10 +218,15 @@ def _transform(data: t.Any) -> t.Any:
             elif isinstance(data, (list, tuple)):
                 data = type(data)([_transform(v) for v in data])
                 types = set(type(i) for i in data)
-                if len(types) > 1:
-                    return Sequence(data=data, auto_convert=True)
-                else:
+                if (
+                    not types
+                    or len(types) == 1
+                    and check_python_scalar_type(types.pop())
+                ):
+                    # Only keep the simple format(the scalar type in list/tuple) the original type for dataset viewer, such as: [1,2,3]
                     return data
+                else:
+                    return Sequence(data=data, auto_convert=True)
```

Comment:
With this logic, the `[[1, 2], [3, 4]]` will use `Sequence`, is this situation as expected?

Reply:
yes